woocommerce/woocommerce
# WooCommerce Docs: Add support for more core block conversion

## Changes

This PR introduces more core block support to the Markdown ingestion plugin's conversions.

This includes support for:

* code block
* quote block
* table block

There is basic fixtures based supporting tests as well to ensure that conversions look as they should.

## Testing

Before the testing instructions, you'll need to generate a manifest pointing to files from this PR:

1. in this branch navigate to `tools/monorepo-utils` and run `pnpm build`
2. navigate to root of repo and run `pnpm utils md-docs create ./plugins/woocommerce-docs/example-docs woodocs --outputFilePath ./plugins/woocommerce-docs/scripts/manifest.json --baseUrl https://raw.githubusercontent.com/woocommerce/woocommerce/df3543b21cabc2765406b733959d48d3fadc2abc`

Now you can test the manifest processing:

1. Have the `plugins/woocommerce-docs` plugin mounted in your dev environment.
2. Build the docs plugin, via pnpm start
3. Now visit the menu item for the plugin in wp-admin (called WooCommerce Docs)
4. Add a new manifest URL via the basic UI, it will need to point to the sample manifest that you generated.
Assuming you're using docker, The URL should be like http://host.docker.internal:PORTNUMBER/wp-content/plugins/woocommerce-docs/scripts/manifest.json 
5. Its not always easy to get Action Scheduler jobs to run in dev, so you can run them manually via Tools > Scheduled Actions. Find the `woocommerce_docs_manifest_job` task and click "run"
6. After it runs, you should be able to visit `Posts` and see all the generated posts, especially the local development post which should contain examples of all the new block conversions.

<img width="708" alt="Screenshot 2023-07-20 at 4 59 02 PM" src="https://github.com/woocommerce/woocommerce/assets/1281828/874e3112-69f9-4e2f-9689-e1174db30502">
